### DISCUSSION TITLE
Can't Upgrade past 2.3.100/ so-dockerregistry missing

### DISCUSSION THREAD

**Discussion by 8inaryMata1eao** [Security-Onion-Solutions/securityonion#8013] [Unsupported Versions] [answered]
Distributed Deployment
Manager: 1
Search Node: 1 
Sensors: 5
Version: 2.3.100

Over the past few months we ran into an issue with a distributed the deployment. After multiple reboots "so-status" shows everything "WAIT_START" except for Docker, so-aptcacherng and so-kratos.  When running soup from the manager we get errors stating "No such container: so-dockerregistry", see second image.  From previous issues I know to run "so-docker-refresh" that pulls an error where it shows connection refused on local host "Get http://[manager]:5000/v2/: dial tcp 127.0.0.1:5000: connect: connection refused", because there is no so-dockerregistry container running. Then a second error when running docker pull during the execution of that script for "ghcr.io/security-onion-solutions/so-idh:2.3.100".  Which I did check there is no tagged docker image for so-idh at 2.3.100.  Any help would be much appreciated as if I cannot salvage this then the entire installation will be scrapped.  Thanks all.

so-status
![image](https://user-images.githubusercontent.com/80425093/170497956-b33e7e56-3a13-48bc-a87f-e0956d328e96.png)

soup
![image](https://user-images.githubusercontent.com/80425093/170498516-bf08e531-62f3-4636-9d47-713b60d613b5.png)

so-docker-refresh 1
![image](https://user-images.githubusercontent.com/80425093/170499674-0ae3522f-e58c-4b42-bdcb-7d2472a61f4e.png)

so-docker-refresh 2 (Yes it can get to the domain and path)
![image](https://user-images.githubusercontent.com/80425093/170499794-feb7ce19-9b8c-4853-8b2f-381f892907b3.png)

Git Hub so-idh Docker Repo (When browsing it appears there is no tag for 2.3.100)
https://github.com/orgs/security-onion-solutions/packages/container/so-idh/versions
![image](https://user-images.githubusercontent.com/80425093/170500210-7813d3f7-f135-4c7d-a47f-cb25d79b51ba.png)

**Comment by dougburks** [comment, marked as answer]
IDH wasn't a part of 2.3.100, it was added in 2.3.110:
https://docs.securityonion.net/en/2.3/idh.html#intrusion-detection-honeypot

What is the output of the following?
```
cat /etc/soversion
```

```
sudo grep soversion /opt/so/saltstack/local/pillar/global.sls
```

**Reply by 8inaryMata1eao** [reply]
Thank you for responding Doug.  Thats why I was confused as to why it was looking for that docker image to pull.

![image](https://user-images.githubusercontent.com/80425093/170719664-d4426a67-bbac-4dce-8792-412bdd14ccc3.png)

**Reply by dougburks** [reply]
Have you checked for additional clues in `/opt/so/log/` specifically `/opt/so/log/docker-registry/`?

Have you checked the output of `sudo docker logs so-dockerregistry`?

**Reply by 8inaryMata1eao** [reply]
Yeah I tried that before I made the discussion.  I'm not sure if you saw the second image with the output of soup.
"error response from daemon no such container so-dockerregistry"

Docker registry does not exist.
![image](https://user-images.githubusercontent.com/80425093/170781028-c7f80fa2-6cd5-41f7-bfd4-e887cff76d13.png)

How would I get that up and running?

**Reply by palichx** [reply]
Hi,
Compare salt-minion version in system and in file
/opt/so/saltstack/default/salt/salt/minion.defaults.yaml

**Reply by 8inaryMata1eao** [reply]
Thanks for the follow up.  I did it appears they are both 3004. This is on the manager.

![image](https://user-images.githubusercontent.com/80425093/171009890-782967a2-b530-4537-938e-8893969a0e25.png)

**Reply by palichx** [reply]
it's a problem. change it to 3004.1(in file)
![image](https://user-images.githubusercontent.com/80513304/171016736-902ea8f5-a98f-4f0a-a32c-4a112fc455ae.png)

**Reply by 8inaryMata1eao** [reply]
Thanks for the tip.  I added the .1 and rebooted
Re-ran soup and tried so-docker-refresh.  Still the same issue.  It appears its all from not being able to access the  so-docker-registry container.

**Reply by palichx** [reply]
hm, i think problem in salt yaml :)
check /opt/so/saltstack/default/salt/salt/master.defaults.yaml

**Reply by 8inaryMata1eao** [reply]
Yup "3004" not "3004.1".  You were exactly correct. After changing both yaml files to 3004.1 rebooting then soup + so-docker-refresh.  Everything was operational.
